src/utils/__tests__/initilizerTests.js

```diff
@@ -0,0 +1,362 @@
+  assert.truthy(res);
+
+  assert.deepEqual(res.initialState, {
+    ...expectedDefaultInitialState,
+
+    defaults: true,
+    user: true,
+    plugin: 1,
+  });
+});
+
+test('init returns composed reducer given plugins', (assert) => {
+  const ctx = {
+    props: {
+      plugins: [
+        { reducer: { PLUGIN: () => ({ plugin: 0 }) } },
+        { reducer: { PLUGIN: () => ({ plugin: 1 }) } },
+      ],
+    },
+  };
+  const defaults = {
+    reducers: {
+      DEFAULTS: () => ({ defaults: true }),
+      PLUGIN: () => ({ plugin: false }),
+    },
+  };
+
+  const res = init.call(ctx, defaults);
+  assert.truthy(res);
+
+  assert.is(typeof res.reducers, 'function');
+  assert.deepEqual(Object.keys(res.reducers), ['DEFAULTS', 'PLUGIN']);
+  assert.deepEqual(res.reducers({}, { type: 'DEFAULTS' }), { defaults: true });
+  assert.deepEqual(res.reducers({}, { type: 'PLUGIN' }), { plugin: 1 });
+});
+
+test('init returns flattened/compacted reduxMiddleware given plugins', (assert) => {
+  const mw = _.range(0, 4).map(i => () => i);
+  const ctx = {
+    props: {
+      plugins: [
+        {},
+        { reduxMiddleware: [mw[0]] },
+        {},
+        { reduxMiddleware: [null, mw[1], undefined, mw[2], null] },
+        {},
+      ],
+      reduxMiddleware: [null, mw[3], undefined],
+    },
+  };
+  const defaults = {};
+
+  const res = init.call(ctx, defaults);
+  assert.truthy(res);
+
+  assert.deepEqual(res.reduxMiddleware, mw);
+});
+
+test('init sets context.components as expected given plugins', (assert) => {
+  const ctx = {
+    props: {
+      plugins: [
+        { components: { Plugin: 0, User: false } },
+        { components: { Plugin: 1 } },
+      ],
+      components: { User: true },
+    },
+  };
+  const defaults = {
+    components: {
+      Defaults: true,
+      Plugin: false,
+    },
+  };
+
+  const res = init.call(ctx, defaults);
+  assert.truthy(res);
+
+  assert.deepEqual(ctx.components, {
+    Defaults: true,
+    Plugin: 1,
+    User: true,
+  });
+});
+
+test('init sets context.settingsComponentObjects as expected given plugins', (assert) => {
+  const ctx = {
+    props: {
+      plugins: [
+        { settingsComponentObjects: { Plugin: 0, User: false } },
+        { settingsComponentObjects: { Plugin: 1 } },
+      ],
+      settingsComponentObjects: { User: true },
+    },
+  };
+  const defaults = {
+    settingsComponentObjects: {
+      Defaults: true,
+      Plugin: false,
+    },
+  };
+
+  const res = init.call(ctx, defaults);
+  assert.truthy(res);
+
+  assert.deepEqual(ctx.settingsComponentObjects, {
+    Defaults: true,
+    Plugin: 1,
+    User: true,
+  });
+});
+
+test('init sets context.events as expected given plugins', (assert) => {
+  const ctx = {
+    props: {
+      plugins: [
+        { events: { Plugin: 0, User: false } },
+        { events: { Plugin: 1 } },
+      ],
+      events: { User: true, User2: true },
+    },
+  };
+  const defaults = {
+    // TODO: bug that defaultEvents is not used?
```

Comment:
@joellanciaux unclear from history, but I'm pretty sure you created `defaultEvents` but it was never used. Thoughts?

Reply:
Yikes.  I'm having a hard time following what was happening going off the `blame`. 😅 

I _think_ it was related to some composition updates I was wanting at the time, but I'm positive it's a perfect candidate for removal.

Reply:
✅ https://github.com/GriddleGriddle/Griddle/pull/757/commits/764a47edddd302a6f66840f7dd5f1190f5493c45